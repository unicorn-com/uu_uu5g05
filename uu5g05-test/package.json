{
  "name": "uu5g05-test",
  "version": "1.40.1",
  "description": "The library contains tools for testing of a components.",
  "license": "GPL-3.0",
  "namespace": "Uu5Test",
  "spec": "nodejs-lib; uu_appg01_devkit@6.6.0; uu5devkitg01-plugin@1.5.1",
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.6.3",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.6.1",
    "fake-indexeddb": "6.0.0",
    "jest-localstorage-mock": "2.4.26",
    "node-fetch": "2.6.7",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "react-test-renderer": "18.3.1"
  },
  "devDependencies": {
    "eslint-plugin-jest-dom": "5.5.0",
    "eslint-plugin-testing-library": "6.5.0",
    "nyc": "15.1.0",
    "uu_appg01_devkit": "^6.6.0",
    "uu5devkitg01-plugin": "^1.5.1"
  },
  "engines": {
    "node": ">=12.0.0"
  },
  "scripts": {
    "clean": "uu_appg01_devkit clean",
    "test": "uu_appg01_devkit testBrowser",
    "build": "uu_appg01_devkit build",
    "package": "uu_appg01_devkit package",
    "start": "uu_appg01_devkit start",
    "prettify": "uu_appg01_devkit prettify",
    "vulnerabilityScan": "uu_appg01_devkit vulnerabilityScan",
    "uuAppDevKitUpgrade": "uu_appg01_devkit uuAppDevKitUpgrade",
    "uu5Generate": "uu5devkitg01-plugin generate"
  },
  "main": "src/index.js",
  "files": [
    "src",
    "dist*",
    "test/setup"
  ],
  "prettier": {
    "printWidth": 120
  }
}
