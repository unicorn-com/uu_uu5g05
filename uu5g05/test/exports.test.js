import Uu5, { Utils } from "uu5g05";
import { omitConsoleLogs } from "uu5g05-test";

const memo = Utils.Component.memo(() => {});

beforeEach(() => {
  omitConsoleLogs(`This class is deprecated. Use Utils.Uu5ObjectStore.Dao instead.`);
});

describe(`Uu5`, () => {
  it("exported API", async () => {
    let apiMap = {};
    addKeys(Uu5, apiMap);
    expect(apiMap).toMatchSnapshot();
  });
});

function addKeys(obj, result, filter = (key, obj, prefix) => true, prefix = "", addedSet = new Set()) {
  if (prefix) result[prefix] = typeof obj;
  if (!obj || (typeof obj !== "object" && typeof obj !== "function")) return;
  if (addedSet.has(obj)) return;
  addedSet.add(obj);
  for (let k of Object.getOwnPropertyNames(obj)) {
    if (/^_/.test(k)) continue;
    if (
      /^(defaultProps|displayName|isStateless|isUu5PureComponent|length|logger|name|nestingLevel|propTypes|prototype|uu5ComponentType|uu5Tag|getDerivedStateFromError|getDerivedStateFromProps)$/.test(
        k,
      )
    ) {
      continue;
    }
    if (obj.$$typeof === memo.$$typeof && k in memo) continue;
    if (!filter(k, obj, prefix)) continue;
    let childFilter;
    if (/^[a-z]/.test(k) || /^[A-Z_0-9]+$/.test(k)) childFilter = () => false;
    else if (/^[A-Z]/.test(k)) childFilter = (k, obj) => k in obj;
    else continue;
    addKeys(obj[k], result, childFilter, prefix ? prefix + "." + k : k, addedSet);
  }
}
