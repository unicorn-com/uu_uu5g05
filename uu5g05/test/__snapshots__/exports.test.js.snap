// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uu5 exported API 1`] = `
{
  "AnimationLayerProvider": "function",
  "AnimationLayerProvider.BUSY_ERROR": "string",
  "AnimationLayerProvider.CANCEL_ERROR": "string",
  "AnimationLayerProvider.MAX_COUNT_1MIN_ERROR": "string",
  "AnimationLayerProvider.MAX_COUNT_5MIN_ERROR": "string",
  "AnimationLayerProvider.NOT_ALLOWED_ERROR": "string",
  "AnimationLayerProvider.NO_PROVIDER_ERROR": "string",
  "AnimationLayerProvider.REPLACE_ERROR": "string",
  "AppBackgroundProvider": "function",
  "AutoLoad": "function",
  "BackgroundProvider": "function",
  "Content": "function",
  "ContentSizeProvider": "function",
  "DataControllerProvider": "function",
  "DataControllerProvider.create": "function",
  "DataStoreProvider": "function",
  "DeviceProvider": "function",
  "DeviceProvider.device": "object",
  "DynamicLibraryComponent": "function",
  "Environment": "object",
  "Environment.appBaseUri": "string",
  "Environment.componentUveUri": "string",
  "Environment.get": "function",
  "Environment.isMobileApp": "boolean",
  "Environment.isSimpleRender": "boolean",
  "Environment.libraryRegistryBaseUri": "string",
  "Environment.plus4UGoBaseUri": "string",
  "Environment.trustedUriRegExp": "string",
  "Environment.uu5DataMap": "object",
  "ErrorBoundary": "function",
  "Fragment": "symbol",
  "LanguageListProvider": "function",
  "LanguageProvider": "function",
  "LevelProvider": "function",
  "Lsi": "function",
  "PortalElementProvider": "function",
  "PropTypes": "object",
  "PropTypes.COLOR_SCHEME": "object",
  "PropTypes.COLOR_SCHEME_LIST": "object",
  "PropTypes.PropTypes": "object",
  "PropTypes.PropTypes.PropTypes": "object",
  "PropTypes.PropTypes.any": "function",
  "PropTypes.PropTypes.array": "function",
  "PropTypes.PropTypes.arrayOf": "function",
  "PropTypes.PropTypes.bool": "function",
  "PropTypes.PropTypes.checkPropTypes": "function",
  "PropTypes.PropTypes.element": "function",
  "PropTypes.PropTypes.elementType": "function",
  "PropTypes.PropTypes.exact": "function",
  "PropTypes.PropTypes.func": "function",
  "PropTypes.PropTypes.instanceOf": "function",
  "PropTypes.PropTypes.node": "function",
  "PropTypes.PropTypes.number": "function",
  "PropTypes.PropTypes.object": "function",
  "PropTypes.PropTypes.objectOf": "function",
  "PropTypes.PropTypes.oneOf": "function",
  "PropTypes.PropTypes.oneOfType": "function",
  "PropTypes.PropTypes.resetWarningCache": "function",
  "PropTypes.PropTypes.shape": "function",
  "PropTypes.PropTypes.string": "function",
  "PropTypes.PropTypes.symbol": "function",
  "PropTypes.any": "function",
  "PropTypes.array": "function",
  "PropTypes.arrayOf": "function",
  "PropTypes.background": "function",
  "PropTypes.bool": "function",
  "PropTypes.borderRadius": "function",
  "PropTypes.checkPropTypes": "function",
  "PropTypes.colorScheme": "function",
  "PropTypes.dataObject": "function",
  "PropTypes.default": "object",
  "PropTypes.element": "function",
  "PropTypes.elementType": "function",
  "PropTypes.exact": "function",
  "PropTypes.func": "function",
  "PropTypes.icon": "function",
  "PropTypes.instanceOf": "function",
  "PropTypes.isDeprecated": "function",
  "PropTypes.isRequiredIf": "function",
  "PropTypes.lsi": "function",
  "PropTypes.lsiMap": "function",
  "PropTypes.node": "function",
  "PropTypes.number": "function",
  "PropTypes.object": "function",
  "PropTypes.objectOf": "function",
  "PropTypes.oneOf": "function",
  "PropTypes.oneOfType": "function",
  "PropTypes.resetWarningCache": "function",
  "PropTypes.shape": "function",
  "PropTypes.sizeOf": "function",
  "PropTypes.space": "function",
  "PropTypes.string": "function",
  "PropTypes.symbol": "function",
  "PropTypes.unit": "function",
  "RenderCounter": "function",
  "RouteProvider": "function",
  "ScreenSizeProvider": "function",
  "SelectionProvider": "function",
  "SessionProvider": "function",
  "Suspense": "symbol",
  "TimeZoneProvider": "function",
  "ToolbarProvider": "function",
  "UserPreferencesProvider": "function",
  "Utils": "object",
  "Utils.Clipboard": "object",
  "Utils.Clipboard.read": "function",
  "Utils.Clipboard.write": "function",
  "Utils.Color": "object",
  "Utils.Color.isLight": "function",
  "Utils.Color.toHex": "function",
  "Utils.Color.toRgba": "function",
  "Utils.Component": "object",
  "Utils.Component.combineRefs": "function",
  "Utils.Component.createRef": "function",
  "Utils.Component.forwardRef": "function",
  "Utils.Component.getUu5StringProps": "function",
  "Utils.Component.lazy": "function",
  "Utils.Component.memo": "function",
  "Utils.Component.mergeStatics": "function",
  "Utils.Component.toUu5String": "function",
  "Utils.Content": "object",
  "Utils.Content.build": "function",
  "Utils.Content.count": "function",
  "Utils.Content.forEach": "function",
  "Utils.Content.map": "function",
  "Utils.Content.only": "function",
  "Utils.Content.toArray": "function",
  "Utils.Context": "object",
  "Utils.Context.create": "function",
  "Utils.Css": "object",
  "Utils.Css.createCssModule": "function",
  "Utils.Css.joinClassName": "function",
  "Utils.Dom": "object",
  "Utils.Dom.addCss": "function",
  "Utils.Dom.createPortal": "function",
  "Utils.Dom.findNode": "function",
  "Utils.Dom.flushSync": "function",
  "Utils.Dom.hydrate": "function",
  "Utils.Dom.render": "function",
  "Utils.Dom.unmount": "function",
  "Utils.Element": "object",
  "Utils.Element.clone": "function",
  "Utils.Element.create": "function",
  "Utils.Element.isValid": "function",
  "Utils.Error": "object",
  "Utils.Error.Message": "function",
  "Utils.Error.addMetadata": "function",
  "Utils.Event": "function",
  "Utils.EventManager": "object",
  "Utils.EventManager.register": "function",
  "Utils.EventManager.trigger": "function",
  "Utils.EventManager.unregister": "function",
  "Utils.Function": "object",
  "Utils.Function.debounce": "function",
  "Utils.Function.memo": "function",
  "Utils.Language": "object",
  "Utils.Language.getItem": "function",
  "Utils.Language.getLanguage": "function",
  "Utils.Language.setLanguage": "function",
  "Utils.LibraryRegistry": "object",
  "Utils.LibraryRegistry.getComponentByUu5Tag": "function",
  "Utils.LibraryRegistry.getLibrary": "function",
  "Utils.LibraryRegistry.importLibrary": "function",
  "Utils.LibraryRegistry.listLibraries": "function",
  "Utils.LibraryRegistry.registerComponent": "function",
  "Utils.LibraryRegistry.registerLibrary": "function",
  "Utils.LoggerFactory": "function",
  "Utils.LoggerFactory.get": "function",
  "Utils.LoggerFactory.init": "function",
  "Utils.Lsi": "function",
  "Utils.Lsi.config": "object",
  "Utils.Lsi.default": "object",
  "Utils.Lsi.getMessage": "function",
  "Utils.Lsi.setDefaultLsi": "function",
  "Utils.Lsi.store": "object",
  "Utils.NestingLevel": "object",
  "Utils.NestingLevel.buildList": "function",
  "Utils.NestingLevel.compare": "function",
  "Utils.NestingLevel.getChildNestingLevel": "function",
  "Utils.NestingLevel.getNestingLevel": "function",
  "Utils.NestingLevel.valueList": "object",
  "Utils.Number": "object",
  "Utils.Number.format": "function",
  "Utils.Number.parse": "function",
  "Utils.Number.random": "function",
  "Utils.Object": "object",
  "Utils.Object.deepEqual": "function",
  "Utils.Object.matchByKey": "function",
  "Utils.Object.mergeDeep": "function",
  "Utils.Object.shallowEqual": "function",
  "Utils.Object.toFlatParams": "function",
  "Utils.ObjectStore": "function",
  "Utils.ObjectStore.Dao": "function",
  "Utils.ObjectStore.Error": "function",
  "Utils.ObjectStore.Error.DuplicateKey": "function",
  "Utils.ObjectStore.Error.ERROR_PREFIX": "string",
  "Utils.ObjectStore.Error.InvalidCall": "function",
  "Utils.ObjectStore.Error.InvalidRevision": "function",
  "Utils.ObjectStore.Error.MissingRevision": "function",
  "Utils.ObjectStore.Error.ObjectNotFound": "function",
  "Utils.Print": "object",
  "Utils.Print.registerPrintBlocker": "function",
  "Utils.Print.unregisterPrintBlocker": "function",
  "Utils.ScreenSize": "object",
  "Utils.ScreenSize.L": "number",
  "Utils.ScreenSize.M": "number",
  "Utils.ScreenSize.S": "number",
  "Utils.ScreenSize.XL": "number",
  "Utils.ScreenSize.XS": "number",
  "Utils.ScreenSize.convertObjectToString": "function",
  "Utils.ScreenSize.convertStringToObject": "function",
  "Utils.ScreenSize.countSize": "function",
  "Utils.ScreenSize.getSize": "function",
  "Utils.ScreenSize.getSizeValue": "function",
  "Utils.ScreenSize.parseValue": "function",
  "Utils.Speech": "object",
  "Utils.Speech.cancel": "function",
  "Utils.Speech.getVoices": "function",
  "Utils.Speech.isEnabled": "boolean",
  "Utils.Speech.onVoicesChanged": "function",
  "Utils.Speech.pause": "function",
  "Utils.Speech.paused": "undefined",
  "Utils.Speech.pending": "undefined",
  "Utils.Speech.resume": "function",
  "Utils.Speech.speak": "function",
  "Utils.Speech.speaking": "undefined",
  "Utils.String": "object",
  "Utils.String.capitalize": "function",
  "Utils.String.format": "function",
  "Utils.String.generateId": "function",
  "Utils.String.stripAccents": "function",
  "Utils.Style": "object",
  "Utils.Style.getMaxMediaQueries": "function",
  "Utils.Style.getMediaQueries": "function",
  "Utils.Style.getMinMediaQueries": "function",
  "Utils.Style.parse": "function",
  "Utils.Style.parseSpace": "function",
  "Utils.Style.replaceAdaptiveSpacing": "function",
  "Utils.Telemetry": "function",
  "Utils.Telemetry.error": "function",
  "Utils.Telemetry.info": "function",
  "Utils.Telemetry.init": "function",
  "Utils.Telemetry.warn": "function",
  "Utils.Uu5Data": "function",
  "Utils.Uu5Data.parse": "function",
  "Utils.Uu5Json": "function",
  "Utils.Uu5Json.parse": "function",
  "Utils.Uu5Json.toJson": "function",
  "Utils.Uu5Loader": "object",
  "Utils.Uu5Loader.Uu5Loader": "object",
  "Utils.Uu5Loader.addEventListener": "function",
  "Utils.Uu5Loader.config": "function",
  "Utils.Uu5Loader.get": "function",
  "Utils.Uu5Loader.import": "function",
  "Utils.Uu5Loader.initUuApp": "function",
  "Utils.Uu5Loader.initUuAppResultState": "undefined",
  "Utils.Uu5Loader.namespace": "string",
  "Utils.Uu5Loader.refreshCache": "function",
  "Utils.Uu5Loader.removeEventListener": "function",
  "Utils.Uu5Loader.resolve": "function",
  "Utils.Uu5Loader.version": "string",
  "Utils.Uu5ObjectStore": "function",
  "Utils.Uu5ObjectStore.Dao": "function",
  "Utils.Uu5ObjectStore.Errors": "object",
  "Utils.Uu5ObjectStore.Errors.AmbiguousQueryError": "function",
  "Utils.Uu5ObjectStore.Errors.ConcurrentModificationError": "function",
  "Utils.Uu5ObjectStore.Errors.CreateManyFailedError": "function",
  "Utils.Uu5ObjectStore.Errors.InsertManyFailedError": "function",
  "Utils.Uu5ObjectStore.Errors.InvalidCallError": "function",
  "Utils.Uu5ObjectStore.Errors.MissingAttributeError": "function",
  "Utils.Uu5ObjectStore.Errors.NotUniqueError": "function",
  "Utils.Uu5ObjectStore.Errors.ObjectStoreError": "function",
  "Utils.Uu5ObjectStore.Errors.ObjectStoreError.ERROR_PREFIX": "string",
  "Utils.Uu5ObjectStore.Errors.ReplaceManyFailedError": "function",
  "Utils.Uu5ObjectStore.Errors.UnexpectedError": "function",
  "Utils.Uu5ObjectStore.Errors.UpdateManyFailedError": "function",
  "Utils.Uu5String": "function",
  "Utils.Uu5String.Object": "function",
  "Utils.Uu5String.Object.create": "function",
  "Utils.Uu5String.Props": "function",
  "Utils.Uu5String.contentToChildren": "function",
  "Utils.Uu5String.contentToObject": "function",
  "Utils.Uu5String.contentToPlainText": "function",
  "Utils.Uu5String.contentToString": "function",
  "Utils.Uu5String.parse": "function",
  "Utils.Uu5String.parseTagPropsArray": "function",
  "Utils.Uu5String.toChildren": "function",
  "Utils.Uu5String.toObject": "function",
  "Utils.Uu5String.toPlainText": "function",
  "Utils.Uu5String.toString": "function",
  "Utils.VisualComponent": "object",
  "Utils.VisualComponent.getAttrs": "function",
  "Utils.VisualComponent.splitProps": "function",
  "createComponent": "function",
  "createComponentWithRef": "function",
  "createHoc": "function",
  "createVisualComponent": "function",
  "createVisualComponentWithRef": "function",
  "useActiveSubscriber": "function",
  "useAnimationLayer": "function",
  "useAppBackground": "function",
  "useBackground": "function",
  "useCall": "function",
  "useCallback": "function",
  "useComponentVisibility": "function",
  "useContentSize": "function",
  "useContentSizeValue": "function",
  "useContext": "function",
  "useDataController": "function",
  "useDataFilter": "function",
  "useDataList": "function",
  "useDataObject": "function",
  "useDataSelection": "function",
  "useDataSorter": "function",
  "useDataStore": "function",
  "useDebugValue": "function",
  "useDeferredValue": "function",
  "useDevice": "function",
  "useDynamicLibraryComponent": "function",
  "useEffect": "function",
  "useElementSize": "function",
  "useEvent": "function",
  "useId": "function",
  "useImperativeHandle": "function",
  "useInsertionEffect": "function",
  "useLanguage": "function",
  "useLanguageList": "function",
  "useLayoutEffect": "function",
  "useLevel": "function",
  "useLsi": "function",
  "useLsiValues": "function",
  "useMemo": "function",
  "useMemoObject": "function",
  "usePreviousValue": "function",
  "usePrint": "function",
  "usePrintBlocker": "function",
  "useReducer": "function",
  "useRef": "function",
  "useRoute": "function",
  "useRouteBack": "function",
  "useRouteLeave": "function",
  "useRouteParams": "function",
  "useRouter": "function",
  "useScreenSize": "function",
  "useScrollDirection": "function",
  "useSelection": "function",
  "useSession": "function",
  "useSessionUntrustedData": "function",
  "useSlide": "function",
  "useSpeechToText": "function",
  "useSpeechVoices": "function",
  "useState": "function",
  "useStickyTop": "function",
  "useSyncExternalStore": "function",
  "useTextToSpeech": "function",
  "useTimeZone": "function",
  "useToolbar": "function",
  "useTraceUpdate": "function",
  "useTransition": "function",
  "useUnmountedRef": "function",
  "useUpdateEffect": "function",
  "useUpdateLayoutEffect": "function",
  "useUserPreferences": "function",
  "useUveVisibility": "function",
  "useValueChange": "function",
  "useViewportVisibility": "function",
  "useWillMount": "function",
  "useWorker": "function",
  "withLazy": "function",
  "withLsi": "function",
  "withRouteParamsProvider": "function",
  "withStickyTop": "function",
  "withViewportStickyBottom": "function",
}
`;
